src/dftbp/dftb/repulsive/chimesrep.F90

```diff
@@ -0,0 +1,317 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2020  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+
+!> Implements a repulsive correction using the ChIMES force field
+module dftbp_dftb_repulsive_chimesrep
+  use dftbp_common_accuracy, only : dp
+  use dftbp_common_constants, only : AA__Bohr, Bohr__AA, kcal_mol__Hartree, Hartree__kcal_mol
+  use dftbp_dftb_periodic, only : TNeighbourList
+  use dftbp_dftb_repulsive_repulsive, only : TRepulsive
+  use dftbp_extlibs_chimes, only : TChimesCalc, TChimesCalc_init
+  implicit none
+
+  private
+  public :: TChimesRepInp, TChimesRep, TChimesRep_init
+
+
+#:if WITH_CHIMES
+
+  !> Input for the Chimes repulsive calcualtor
+  type :: TChimesRepInp
+
+    !> Name of the Chimes input file
+    character(:), allocatable :: chimesFile
+
+  end type TChimesRepInp
+
+
+  !> Contains the necessary data to query the ChiIMES force field
+  type, extends(TRepulsive) :: TChimesRep
+    private
+
+    !> Chimes calculator
+    type(TChimesCalc) :: chimesCalc
+
+    !> Lattice vector buffer
+    real(dp) :: latVecs(3, 3) = 0.0_dp
+
+    !> Energy per atom buffer. Shape: [nAtom]
+    real(dp), allocatable :: energyPerAtom(:)
+
+    !> Gradient buffer. Shape: [3, nAtom]
+    real(dp), allocatable :: gradients(:,:)
+
+    !> Stress buffer
+    real(dp) :: stress(3, 3) = 0.0_dp
+
+  contains
+
+    procedure :: getRCutOff => TChimesRep_getRCutOff
+    procedure :: updateLatVecs => TChimesRep_updateLatVecs
+    procedure :: updateCoords => TChimesRep_updateCoords
+    procedure :: getEnergy => TChimesRep_getEnergy
+    procedure :: getGradients => TChimesRep_getGradients
+    procedure :: getStress => TChimesRep_getStress
+    procedure :: updateSpecies => TChimesRep_updateSpecies
+
+  end type TChimesRep
+
+
+  !> Structure constructor
+  interface TChimesRep
+    module procedure TChimesRep_construct
+  end interface
+
+#:else
+
+  type :: TChimesRepInp
+  end type TChimesRepInp
+
+  type :: TChimesRep
+  end type TChimesRep
```

Comment:
I think it is preferable to remove the (private) members of the type which are unavailable without Chimes and but keep the type mostly intact.